src/rules/noShadowedVariableRule.ts

```diff
@@ -33,122 +36,257 @@ export class Rule extends Lint.Rules.AbstractRule {
     };
     /* tslint:enable:object-literal-sort-keys */
 
-    public static FAILURE_STRING_FACTORY = (name: string) => {
+    public static FAILURE_STRING(name: string): string {
         return `Shadowed variable: '${name}'`;
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithWalker(new NoShadowedVariableWalker(sourceFile, this.getOptions()));
+        return this.applyWithWalker(new Walker(sourceFile, this.ruleName, undefined));
     }
 }
 
-class NoShadowedVariableWalker extends Lint.BlockScopeAwareRuleWalker<Set<string>, Set<string>> {
-    public createScope() {
-        return new Set();
-    }
+class Walker extends Lint.AbstractWalker<void> {
```

Comment:
Don't get me wrong, but to me this implementation seems to be overly complex for this use case.
Do you mind if I try to come up with a simpler approach?